Modules/Filtering/LabelMap/include/itkStatisticsLabelMapFilter.hxx

```diff
@@ -177,15 +185,28 @@ StatisticsLabelMapFilter<TImage, TFeatureImage>::ThreadedProcessLabelObject(Labe
   }
 
   // the median
-  double median = 0;
-  double count = 0; // will not be fully set, so do not use later !
+  double median = 0.0;
+  double count = 0.0; // will not be fully set, so do not use later !
   for (SizeValueType i = 0; i < histogram->Size(); ++i)
   {
     count += histogram->GetFrequency(i);
 
-    if (count >= (totalFreq / 2))
+    if (count >= ((totalFreq + 1) / 2))
     {
       median = histogram->GetMeasurementVector(i)[0];
+      // If there are an even number of elements average with the next bin with elements
+      if (labelObject->Size() % 2 == 0 && count == totalFreq / 2)
```

Comment:
Are `labelObject->Size()` and `totalFreq` equal?  If so, why are both forms present here?  Is it that they are different types?